specification/communication/data-plane/Rooms/preview/2022-02-01-preview/communicationservicesrooms.json

```diff
@@ -0,0 +1,846 @@
+            "description": "InternalServerError",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Update meeting room's properties.": {
+            "$ref": "./examples/Rooms_UpdateRoom.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Rooms"
+        ],
+        "summary": "Delete a room.",
+        "operationId": "Rooms_DeleteRoom",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "roomId",
+            "description": "The id of the room to be deleted",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "The room was successfully deleted."
+          },
+          "400": {
+            "description": "Bad Request",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "401": {
+            "description": "Unauthorized",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "403": {
+            "description": "Forbidden",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "500": {
+            "description": "InternalServerError",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Delete a meeting room.": {
+            "$ref": "./examples/Rooms_DeleteRoom.json"
+          }
+        }
+      }
+    },
+    "/rooms/{roomId}/participants": {
+      "get": {
+        "tags": [
+          "Participants"
+        ],
+        "summary": "Get participants in a room.",
+        "operationId": "Rooms_GetParticipants",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "roomId",
+            "description": "The id of the room to get participants from",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The action returns the participants of a room.",
+            "schema": {
+              "$ref": "#/definitions/ParticipantsCollection"
```

Comment:
Do we know maximum number of the participants? Do we need to support pagination?

Reply:
@radubulboaca Adding @radubulboaca to comment here based on what we foresee as we are targeting Public preview in Q2 2022.

Reply:
Thank you for looking into this @weidongxu-microsoft.

For public preview, Rooms service has a soft limit of 350 participants. The main customer scenario targeted for this service are virtual visits (e.g., doctor appointments, financial advisors, classrooms), which inherently have relatively few participants. Because of this, the team made a product decision to not include pagination at this point. As usage patterns change and we start seeing demand for large number of room participants, we plan to add pagination into the API. 

Also notable in this context is the concept of 'open room', which is also present in this API. Marking the room as 'open' allows users to join the room call when they haven't been explicitly added as participants to the room via this API (assuming the users have the proper authentication tokens). As the number of call attendees increases, managing and updating the roster naturally starts to become more cumbersome for customers (regardless of pagination). Because of this natural tendency, we expect users of large rooms to default to using 'open' rooms, which relives the need for pagination.

Reply:
I will note that adding pagination to a GA'ed API is a breaking change. I'd recommend adding pagination from the start for all APIs that return collections that aren't physically bounded by design (an example would be something that returns the names of countries in the world where a service is supported).

Reply:
Can we follow up on the decision around pagination internally?

I am not yet convinced that this service should GA without pagination. The guidelines require pagination before GA because it is often needed after the fact and is a breaking change. We should only GA without pagination if we're saying we will limit the number of participants to a hard limit (of less than, say 350) for the next 3 years.

Reply:
captured pagination to discussions to https://skype.visualstudio.com/SPOOL/_workitems/edit/2930042 to be discussed and worked on incrementally before GA.